internal/storage/elem.vaults.go

```diff
@@ -626,73 +626,84 @@ func GetKongStrategyDebt(chainID uint64, vaultAddress, strategyAddress common.Ad
 }
 
 /**************************************************************************************************
-** FetchAndStoreKongStrategyData ensures kong strategy data is available in cache, fetching if needed
+** FetchAndStoreKongStrategyData ensures kong strategy data is available in cache.
+** Uses collection query to fetch all strategies for the chain at once, which is more efficient
+** than individual queries.
 **************************************************************************************************/
 func FetchAndStoreKongStrategyData(chainID uint64, address common.Address) (models.TKongStrategySchema, error) {
 	if data, ok := GetKongStrategyData(chainID, address); ok {
 		return data, nil
 	}
 
 	client := kong.NewClient()
-	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 45*time.Second)
 	defer cancel()
 
-	strategy, err := client.FetchStrategyDetails(ctx, chainID, address)
+	// Fetch all strategies for the chain in one efficient collection query
+	strategies, err := client.FetchStrategiesForChain(ctx, chainID)
```

Comment:
-- this makes things worse
instead of calling FetchStrategyDetails for each strategy, this calls FetchStrategiesForChain for each strategy. FetchStrategiesForChain should only be called once per chain per indexing cycle.